### DISCUSSION TITLE
Fetch IAP (Recurring) prices and free trial duration with Apple Store Connect API.

### DISCUSSION THREAD

**Discussion by lacett** [fastlane/fastlane#18196] [General]
I have been using Fastlane as part of a fully automated system to build, upload, release and take screenshots of multiple apps. As for iOS 14, IAP have not been loading on simulators and I have to configure the StoreKit files to make it appear. 
I have always used the _Tunes_ module to query IAP's detailed information (specially pricing and free trial duration), but with Apple enforcing 2FA by the end of Feb/21 I won't be able to use it anymore. 

Is there any way I can get these detailed information for IAP's using Apple API Keys and the new modules that communicate with the Apple Store Connect API?

**Comment by dinsen** [comment]
Hi @lacett 

Apple only added one endpoint for IAP's and I don't think you can get the information you want with it:
https://developer.apple.com/documentation/appstoreconnectapi/read_in-app_purchase_information
